src/backend/gporca/libgpopt/src/init.cpp

```diff
@@ -67,15 +56,28 @@ gpopt_init()
 //
 //---------------------------------------------------------------------------
 void
-gpopt_terminate()
+gpopt_terminate(bool shutdown)
 {
 #ifdef GPOS_DEBUG
-	CMDCache::Shutdown();
+	shutdown = true;
+#endif	// GPOS_DEBUG
+
+	if (!shutdown)
+	{
+		return;
+	}
+
+	if (NULL != CMDCache::Pcache())
+	{
+		CMDCache::Shutdown();
+	}
 
 	CMemoryPoolManager::GetMemoryPoolMgr()->Destroy(mp);
```

Comment:
mp may be NULL if malloc at the call stack below returns null.
```
gpos::clib::Malloc (size=304) at clibwrapper.cpp:475
gpos::CMemoryPoolManager::SetupGlobalMemoryPoolManager<gpos::CMemoryPoolPallocManager, gpos::CMemoryPoolPalloc> () at CMemoryPoolManager.h:109
gpos::CMemoryPoolPallocManager::Init () at CMemoryPoolPallocManager.cpp:57
CGPOptimizer::InitGPOPT () at CGPOptimizer.cpp:176
InitGPOPT () at CGPOptimizer.cpp:260
InitPostgres () at postinit.c:664
```
There is GPOS_ASSERT(NULL != mp) at the beginning of CMemoryPoolManager::Destroy. So let's add `if (NULL != mp)` here

Reply:
fixed